With @types/bun since 1.2.7 TypeScript complains about Error.cause/NumberFormatOptions

### What version of Bun is running?

1.2.8+adab0f64f

### What platform is your computer?

Darwin 24.5.0 arm64 arm

### What steps can reproduce the bug?

1. `git clone git@github.com:isnifer/bun-types-issue-127-128.git`
2. `cd bun-types-issue-127-128`
3. `bun install`
4. `bun --filter '*' build`

### What is the expected behavior?

Successful `tsc` check for 1.2.7-1.2.8 like for 1.2.6.
Because the only difference between packages is version of `@types/bun` package.

### What do you see instead?

```sh
bun-127 build $ bunx --bun tsc
│ main.ts(16,25): error TS2550: Property 'cause' does not exist on type 'Error'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2022' or later.
│ ../shared/index.ts(6,43): error TS2554: Expected 0-1 arguments, but got 2.
│ ../shared/index.ts(16,21): error TS2339: Property 'trailingZeroDisplay' does not exist on type 'NumberFormatOptions'.
│ ../shared/index.ts(17,19): error TS2339: Property 'trailingZeroDisplay' does not exist on type 'NumberFormatOptions'.
└─ Exited with code 2
bun-128 build $ bunx --bun tsc
│ main.ts(16,25): error TS2550: Property 'cause' does not exist on type 'Error'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2022' or later.
│ ../shared/index.ts(6,43): error TS2554: Expected 0-1 arguments, but got 2.
│ ../shared/index.ts(16,21): error TS2339: Property 'trailingZeroDisplay' does not exist on type 'NumberFormatOptions'.
│ ../shared/index.ts(17,19): error TS2339: Property 'trailingZeroDisplay' does not exist on type 'NumberFormatOptions'.
└─ Exited with code 2
```

### Additional information

I have one more issue related to `import.meta.env.VAR_NAME` since `@types/bun@1.2.7` but can't reproduce it yet in the clean repo.

--- a/packages/bun-types/globals.d.ts
+++ b/packages/bun-types/globals.d.ts
@@ -872,7 +872,29 @@ declare class BuildMessage {
   readonly level: "error" | "warning" | "info" | "debug" | "verbose";
 }
 
+interface ErrorOptions {
+  /**
+   * The cause of the error.
+   */
+  cause?: unknown;
+}
+
+interface Error {
+  /**
+   * The cause of the error.
+   */
+  cause?: unknown;
+}
+
 interface ErrorConstructor {
+  new (message?: string, options?: ErrorOptions): Error;
+
+  /**
+   * Check if a value is an instance of Error
+   *
+   * @param value - The value to check
+   * @returns True if the value is an instance of Error, false otherwise
+   */
   isError(value: unknown): value is Error;
 
   /**

--- a/test/integration/bun-types/fixture/globals.ts
+++ b/test/integration/bun-types/fixture/globals.ts
@@ -317,3 +317,21 @@ clearInterval(1);
 clearTimeout(1);
 setImmediate(() => {});
 clearImmediate(1);
+
+const err = new Error("test");
+err.cause = "asdf";
+err.cause = new Error("asdf");
+err.cause = new RangeError("asdf");
+err.cause = new TypeError("asdf");
+err.cause = new URIError("asdf");
+
+expectType<unknown>(err.cause);
+expectType<typeof Error>().is<ErrorConstructor>();
+
+new Error("asdf", {
+  cause: "asdf",
+});
+
+new Error("asdf", {
+  cause: new Error("asdf"),
+});

